android/lib/src/main/java/com/ern/api/impl/core/ElectrodeBaseFragmentDelegate.java

```diff
@@ -0,0 +1,297 @@
+                AppCompatActivity appCompatActivity = (AppCompatActivity) mFragment.getActivity();
+                if (appCompatActivity.getSupportActionBar() == null) {
+                    appCompatActivity.setSupportActionBar(toolbar);
+                } else {
+                    Logger.w(TAG, "Hiding fragment layout toolbar. The Activity already has an action bar setup, consider removing the toolbar from your fragment layout.");
+                    toolbar.setVisibility(View.GONE);
+                }
+            } else {
+                Logger.w(TAG, "Ignoring toolbar, looks like the activity is not an AppCompatActivity. Make sure you configure thr toolbar in your fragments onCreateView()");
+            }
+        }
+    }
+
+    @NonNull
+    private Bundle initialProps(boolean isFragmentBeingReconstructed) {
+        final Bundle props = mFragment.getArguments() == null ? new Bundle() : mFragment.getArguments();
+
+        //NOTE: If/When the system re-constructs a fragment from a previous state a stored Bundle is getting converted to a ParcelableData.
+        //When this bundle is send across React native , RN frameworks WritableArray does not support parcelable conversion.
+        //To avoid this issue we recreate the ErnNavRoute object from the bundle and regenerate a new bundle which again replaces the  ParcelableData with proper bundle object.
+        //Checking for the existence of "path" key since that is the only required property to successfully build an ErnNavRoute object.
+        if (isFragmentBeingReconstructed && props.containsKey("path")) {
+            props.putAll(new ErnNavRoute(props).toBundle());
+        }
+
+        Bundle globalProps = mElectrodeActivityListener.globalProps();
+        if (globalProps != null) {
+            props.putAll(globalProps);
+        }
+
+        return props;
+    }
+
+    @SuppressWarnings("unused")
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+    }
+
+    @SuppressWarnings("unused")
+    @OnLifecycleEvent(Lifecycle.Event.ON_START)
+    public void onStart() {
+    }
+
+    @SuppressWarnings("unused")
+    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
+    public void onResume() {
+    }
+
+    @SuppressWarnings("unused")
+    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
+    public void onPause() {
+    }
+
+    @SuppressWarnings("unused")
+    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+    public void onStop() {
+    }
+
+    @SuppressWarnings("unused")
+    public void onDestroyView() {
+    }
+
+    @SuppressWarnings("unused")
+    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+    @CallSuper
+    public void onDestroy() {
+        Logger.v(TAG, "inside onDestroy");
+        if (mMiniAppView != null) {
+            assert mFragment.getArguments() != null;
+            mElectrodeActivityListener.removeReactNativeView(mMiniAppComponentName, mMiniAppView);
+            mMiniAppView = null;
+            mFragmentConfig = null;
```

Comment:
Question: Is is necessary to set mFragmentConfig to null here? The reason I'm asking is because if you don't set it, it will be possible to make the member variable final!

Reply:
Yeah, done.